app/controllers/discourse_automation/admin_discourse_automation_automations_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module DiscourseAutomation
+  class AdminDiscourseAutomationAutomationsController < ::ApplicationController
+    def index
+      automations = DiscourseAutomation::Automation.all
+      serializer = ActiveModel::ArraySerializer.new(
+        automations,
+        each_serializer: DiscourseAutomation::AutomationSerializer,
+        root: 'automations'
+      ).as_json
+      render_json_dump(serializer)
+    end
+
+    def show
+      automation = DiscourseAutomation::Automation.find(params[:id])
+      render_serialized_automation(automation)
+    end
+
+    def create
+      automation_params = params.require(:automation).permit(:name, :script)
+      automation = DiscourseAutomation::Automation.create!(automation_params)
+      render_serialized_automation(automation)
+    end
+
+    def update
+      automation = DiscourseAutomation::Automation.find(params[:id])
+      automation.update!(request.parameters[:automation].slice(:name, :id, :script))
+
+      trigger_params = request.parameters[:automation][:trigger].slice(:metadata, :name)
+      if automation.trigger
+        automation.trigger.update_with_params(trigger_params)
+      else
+        automation.create_trigger!(trigger_params)
+        automation.trigger.update_with_params(trigger_params)
```

Comment:
Do we need this line when we just created the trigger?

Reply:
no ultimately we don't, this is a hack I made for now, I will refactor the controller, I had to hack it to make the relationships work with frontend, but it's good enough to try it